ex1/functions.py

```diff
@@ -90,26 +118,49 @@ def f():
 # Define a function named args_to_dict that takes three arguments and returns a
 # dictionary with the position of the argument as the key (starting at 0) and
 # the argument as the value.
+def args_to_dict(arg0, arg1, arg2):
+	return {0: arg0, 1: arg1, 2: arg2}
 
 # BONUS: Write a function named args_to_dict_general that does the same for any
 # number of arguments
+def args_to_dict_general(*args):
+	keys = range(0, len(args))
+	return dict(zip(keys, args))
 
 # Define a function named lists_to_dict that takes two lists of equal lenght
 # named keys and values and builds a dictionary out of them.
+def lists_to_dict(keys, values):
+	return dict(zip(keys, values))
+
 
 # Define a function named search_list that takes two lists a and b. The
 # function searches for all elements of b in list a. The return value should be
 # a dictionary containing the index of the found element of b in list a and the
 # value of the found element. If nothing was found then return an empty
 # dictionary.
+def search_list(a, b):
+	d = {}
+	for i in b:
+		if(i in a):
+			d.update({a.index(i): i})
```

Comment:
It is fine like it is but you could also just assign a value to a new key. I find that more readable.

```python
d[a.index(i)] = i
```